iml-system-docker-tests/tests/docker_integration_test.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use std::collections::{HashMap, hash_map::RandomState};
+use iml_system_test_utils::{docker, iml, vagrant, get_local_server_names};
+use iml_systemd;
+use iml_wire_types;
+
+async fn setup() -> Result<(), Box<dyn std::error::Error>> {
+    // remove the stack if it is running and clean up volumes and network
+    docker::remove_iml_stack().await?;
+    docker::system_prune().await?;
+    docker::volume_prune().await?;
+    docker::configure_docker_overrides().await?;
+    iml_systemd::restart_unit("docker.service".into()).await?;
+
+    // Destroy any vagrant nodes that are currently running
+    vagrant::destroy().await?;
+
+    Ok(())
+}
+
+async fn run_fs_test<S: std::hash::BuildHasher>(docker_setup: &docker::DockerSetup, server_map: HashMap<String, &[String], S>, fs_type: vagrant::FsType) -> Result<(), Box<dyn std::error::Error>> {
+    let config = vagrant::ClusterConfig::default();
+    setup().await?;
+    docker::configure_docker_setup(&docker_setup).await?;
+
+    docker::deploy_iml_stack().await?;
+
+    vagrant::setup_deploy_docker_servers(
+        &config,
+        server_map
+    ).await?;
+    vagrant::configure_ntp_for_host_only_if(&config).await?;
+    vagrant::create_fs(fs_type, &config.storage_servers()[..]).await?;
+
+    iml::detect_fs().await?;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_docker_ldiskfs_setup() -> Result<(), Box<dyn std::error::Error>> {
+    let server_names = get_local_server_names(&config.storage_servers());
```

Comment:
where is config supposed to come from?

Reply:
Ah now I understand the compiler error you were seeing.